packages/@dcl/ecs/src/systems/events.ts

```diff
@@ -99,8 +99,23 @@ export function createPointerEventSystem(engine: IEngine, inputSystem: IInputSys
 
   function setPointerEvent(entity: Entity, type: PointerEventType, opts: EventSystemOptions) {
     if (opts.hoverText || opts.showFeedback) {
-      const pointerEvent = PointerEvents.getMutableOrNull(entity) || PointerEvents.create(entity)
+      const existingPointerEvent = PointerEvents.getMutableOrNull(entity)
+      //
+      if (existingPointerEvent) {
+        const alreadyExist = existingPointerEvent.pointerEvents.find(
+          (item) =>
+            item.eventType === type &&
+            item.eventInfo &&
+            item.eventInfo.button === opts.button &&
```

Comment:
I think we could only find by button, then mutate it
```suggestion
            item.eventInfo.button === opts.button &&
```